### ISSUE TITLE
getting an error while install kamailio5.1 on ubuntu 18.4

### ISSUE THREAD

**Issue by limitall** [kamailio/kamailio#1602]
flex -o core/lex.yy.c core/cfg.lex
CC (gcc) [kamailio]		core/lex.yy.o
CC (gcc) [kamailio]		core/cfg.tab.o
LD (gcc) [kamailio]		kamailio
CC (gcc) [M db_mysql.so]		km_val.o
CC (gcc) [M db_mysql.so]		my_fld.o
In file included from my_fld.c:22:0:
my_fld.h:37:2: error: unknown type name ‘my_bool’
  my_bool is_null;
  ^~~~~~~
../../Makefile.rules:100: recipe for target 'my_fld.o' failed
make[2]: *** [my_fld.o] Error 1
Makefile:511: recipe for target 'modules' failed
make[1]: *** [modules] Error 1
make[1]: Leaving directory '/usr/local/src/kamailio-5.1/kamailio/src'
Makefile:27: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by linuxmaniac** [comment]
We are building master branch every night for [bionic](https://kamailio.sipwise.com/job/kamailiodev-nightly-binaries/architecture=amd64,distribution=bionic,label=slave/1150/consoleText) with no problems.

```
/usr/bin/make -C src/ every-module 
make[2]: Entering directory '/tmp/buildd/kamailio-5.2.0~dev6+0~20180726010658.1165+bionic/src'
CC (gcc) [M db_mysql.so]		km_dbase.o
CC (gcc) [M db_mysql.so]		km_my_con.o
CC (gcc) [M db_mysql.so]		km_val.o
CC (gcc) [M db_mysql.so]		my_fld.o
CC (gcc) [M db_mysql.so]		km_row.o
CC (gcc) [M db_mysql.so]		my_con.o
CC (gcc) [M db_mysql.so]		km_res.o
CC (gcc) [M db_mysql.so]		my_cmd.o
CC (gcc) [M db_mysql.so]		db_mysql.o
CC (gcc) [M db_mysql.so]		my_res.o
CC (gcc) [M db_mysql.so]		my_uri.o
CC (gcc) [M db_mysql.so]		km_db_mysql.o
make[4]: 'libsrdb2.so.1.0' is up to date.
make[4]: 'libsrdb1.so.1.0' is up to date.
LD (gcc) [M db_mysql.so]		db_mysql.so
make[2]: Leaving directory '/tmp/buildd/kamailio-5.2.0~dev6+0~20180726010658.1165+bionic/src'
```

And there are no related changes on db_mysql AFAICT I would say you are missing dev libraries

**Comment by miconda** [comment]
I guess it was used with mysql 8.0.1, where my_bool type was removed -- patch above should be the fix, but it would require C99 compatible compiler. If master gets compiled/packaged without issues on the systems we build, then it should be ok.

**Comment by sergey-safarov** [comment]
@limitall could you test commit 1736723cc2a15bb75c26711675712a966161b722 or current master
if now master builds correctly, then please close ticket.

**Comment by henningw** [comment]
Just for reference: the fix referenced above caused a similar issue for MariaDB, it is now fixed in commit 9f459dccf84b in master.
